quickwit/quickwit-serve/src/elasticsearch_api/rest_handler.rs

```diff
@@ -547,13 +549,16 @@ async fn es_compat_index_field_capabilities(
     search_params: FieldCapabilityQueryParams,
     search_body: FieldCapabilityRequestBody,
     search_service: Arc<dyn SearchService>,
+    mut metastore: MetastoreServiceClient,
 ) -> Result<FieldCapabilityResponse, ElasticsearchError> {
+    let indexes_metadata = resolve_index_patterns(&index_id_patterns, &mut metastore).await?;
```

Comment:
```suggestion
    let indexes_metadata = resolve_index_patterns(&index_id_patterns, &mut metastore).await?;
```

I'm really not fond of doubling the calls to `resolve_index_patterns`.
We happen to do one in `root_list_fields` too.

It seems like we pre-build /cache the `ListFieldsEntryResponse` in the packager (and indexing time) so that `root_list_fields(..)` is a light operation.

It would make sense to add the bool property there, in the `ListFieldsEntryResponse`.

```
message ListFieldsEntryResponse {
  ...
  // True means that the field is the timestamp field of all of the requested indexes.
  bool is_timestamp = 8;
}
```

We then need to know how and when we populate it.


Solution 1:

The logical place would be the packager.
Pros: that's where we build the rest. 
Cons: Right now the packager does not have access to the docmapper and does not know what the timestamp field is. We would have to pass that information. It is reasonable however. 
Cons2: the data is not available for existing data.
-----------------

Solution 2:
Keep it set to false in the packager, and actually compute it in the root_list_fields function.
At this point we have access to the index_metadatas already.
Pros: easy
Cons: ugly.